src/storage/sstable.rs

```diff
@@ -126,10 +129,7 @@ impl SSTable {
         Ok(())
     }
 
-    pub fn compact<'a, I: Iterator<Item = &'a Self>>(
-        path: PathBuf,
-        tables: I,
```

Comment:
I couldn't make the signature using Iterator work. Something to do with potentially losing the lifetime of the generic argument K. I like Iterator better though. If you know of a way, please make the change here.

Reply:
i don't know a way! 🙏🏽